File: backend/app/models/external_vehicle.py
"""Lightweight external vehicle records (family/friend reference)."""

from __future__ import annotations

from datetime import datetime
from typing import TYPE_CHECKING

from sqlalchemy import DateTime, ForeignKey, Integer, String, Text
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy.sql import func

from app.database import Base

if TYPE_CHECKING:
    from app.models.user import User


class ExternalVehicle(Base):
    """Vehicle tracked for someone without a household account.

    Distinct from ``vehicles`` — optional VIN for NHTSA lookup only (not authz PK);
    no fleet analytics or reminder packs.
    """

    __tablename__ = "external_vehicles"

    id: Mapped[int] = mapped_column(Integer, primary_key=True, autoincrement=True)
    user_id: Mapped[int] = mapped_column(
        Integer,
        ForeignKey("users.id", ondelete="CASCADE"),
        nullable=False,
        index=True,
    )
    nickname: Mapped[str] = mapped_column(String(100), nullable=False)
    vin: Mapped[str | None] = mapped_column(String(17), nullable=True)
    year: Mapped[int | None] = mapped_column(Integer, nullable=True)
    make: Mapped[str | None] = mapped_column(String(50), nullable=True)
    model: Mapped[str | None] = mapped_column(String(50), nullable=True)
    vehicle_type: Mapped[str | None] = mapped_column(String(30), nullable=True)
    contact_name: Mapped[str | None] = mapped_column(String(100), nullable=True)
    contact_phone: Mapped[str | None] = mapped_column(String(40), nullable=True)
    notes: Mapped[str | None] = mapped_column(Text, nullable=True)
    created_at: Mapped[datetime] = mapped_column(DateTime, server_default=func.now(), nullable=False)
    updated_at: Mapped[datetime | None] = mapped_column(DateTime, onupdate=func.now())

    user: Mapped[User] = relationship("User", foreign_keys="[ExternalVehicle.user_id]")

    def __repr__(self) -> str:
        return f"<ExternalVehicle(id={self.id}, nickname={self.nickname!r})>"

File: frontend/src/types/externalVehicle.ts
export interface ExternalVehicle {
  id: number
  nickname: string
  vin: string | null
  year: number | null
  make: string | null
  model: string | null
  vehicle_type: string | null
  contact_name: string | null
  contact_phone: string | null
  notes: string | null
  created_at: string
  updated_at: string | null
}

export interface ExternalVehicleListResponse {
  vehicles: ExternalVehicle[]
  total: number
}

export type ExternalVehicleInput = {
  nickname: string
  vin?: string | null
  year?: number | null
  make?: string | null
  model?: string | null
  vehicle_type?: string | null
  contact_name?: string | null
  contact_phone?: string | null
  notes?: string | null
}
